mongo/replicated_bootstrap/init.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -e
+
+[[ `hostname` =~ -([0-9]+)$ ]] || exit 1
+ordinal=${BASH_REMATCH[1]}
+
+# Execute user creation and replication setup on the first node only
+if [[ $ordinal -ne 0 ]]; then
+    exit 0
+fi
+
+gosu root mongod --transitionToAuth --clusterAuthMode keyFile --keyFile ${KEY_FILE} --replSet ${REPL_SET} --fork --logpath ${DB_ROOT}/init-admin.log
+sleep 30
+
+# check if replication is initialized 
+res="$(mongo --quiet --eval='printjson(rs.status())')"
+if [[ $res != *"NotYetInitialized"* ]]; then
+    exit 0
+fi
+
+name=$(hostname -f)
+
+echo "starting bootstrap"
+mongo --eval "
+rs.initiate( {
+   _id : \"${REPL_SET}\",
+   members: [ { _id : 0, host : \"${name}\" } ]
+})"
+
+# wait for mongodb to understand that it is master
+sleep 10
+
+echo "creating user ${ADMIN_USERNAME}"
+mongo --eval "
+db.createUser({
+  user: \"${ADMIN_USERNAME:?}\",
+  pwd: \"${ADMIN_PASSWORD:?}\",
+  roles: [{ 
+	role: \"root\",
+	db: \"admin\"
+  }]
+});
+" 
+
+mongo --eval "
+db.getSiblingDB(\"admin\").createUser({
+    user: \"${EXPORTER_USERNAME:?}\",
+    pwd: \"${EXPORTER_PASSWORD:?}\",
+    roles: [
+        { role: \"clusterMonitor\", db: \"admin\" },
+        { role: \"read\", db: \"local\" }
+    ]
+});
+"
+
+mongo --eval "
+db.createUser({
+    user: \"${MONGOLIZER_USERNAME:?}\",
+    pwd: \"${MONGOLIZER_PASSWORD:?}\",
+    roles: [{ role: \"backup\", db:\"admin\"}]
+});"
+
+mongo --eval "
+db.createUser({
+    user: \"${SERVICE_USERNAME:?}\",
+    pwd: \"${SERVICE_PASSWORD:?}\",
+    roles: [
+       { role: \"readWrite\", db: \"${SERVICE_DB:?}\" }
+    ]
+});"
+
+echo "rs config init ${name}"
+nodes=$(echo $REPLICATION_NODES | tr "," "\n")
+counter=1
+for node in $nodes
+do
+    echo "adding replica ${node}, ${counter}"
+    while true
+    do
+        script="mongo --quiet --eval 'rs.add({_id: ${counter}, host:\"${node}\", priority: 0.99})'"
+        out=$(eval "$script")
+        echo $out
+        if [[ $out != *"NodeNotFound"* ]]; then
+            break
+        fi
+        echo "retrying ${node}"
+        sleep 5
```

Comment:
This will retry forever pretty quietly right?

Reply:
yeah, mongodb-creator will be stuck on status:`init` and we will have to inspect it via `kubect logs`.  There is no good way around as if I fail here with `exit 1` , k8s will relaunch this and it will pass because replication is already started (`rs.init()`) at this point.

Reply:
Hmm, there's every chance that happens anyway if the k8s node dies or something else outside of your control. What happens/how do we find out about it if it does die and pass the init check?

Reply:
This is executed only one time during creation.

after that this check will say that replication is initialized and no actions will be taken:

```
# check if replication is initialized 
res="$(mongo --quiet --eval='printjson(rs.status())')"
if [[ $res != *"NotYetInitialized"* ]]; then
    exit 0
fi
```